src/data/option/basic.lean

```diff
@@ -87,6 +87,36 @@ by cases a; cases b; refl
 lemma bind_assoc (x : option α) (f : α → option β) (g : β → option γ) :
   (x.bind f).bind g = x.bind (λ y, (f y).bind g) := by cases x; refl
 
+/--
+Flatten an `option` of `option`, a specialization of `mjoin`.
+-/
+@[simp] def join : option (option α) → option α
+| none            := none
+| (some (none))   := none
+| (some (some x)) := some x
```

Comment:
Why not define it as `o.bind id`? Anyway, the last two cases should go together as `| (some x) := x`, then `dsimp` will simplify `join (some x)` to `x`, not only `join (some none)` and `join (some (some x))`.

Reply:
BTW, definition should go to `option/defs`, not `option/basic`.